python/ffsim/hamiltonians/molecular_hamiltonian.py

```diff
@@ -156,6 +158,81 @@ def _fermion_operator_(self) -> FermionOperator:
                 }
             )
         return op
+    
+
+    @staticmethod
+    def from_fermion_operator(op: FermionOperator) -> MolecularHamiltonian:
+        """Convert a FermionOperator to a MolecularHamiltonian."""
+        # extract number of spatial orbitals
+        norb = 1 + max(orb for term in op for _, _, orb in term)
+
+        # initialize constant, one‑ and two‑body tensors
+        constant: float = 0.0
+        one_body_tensor = np.zeros((norb, norb), dtype=complex)
+        two_body_tensor = np.zeros((norb, norb, norb, norb), dtype=complex)
+
+        # track which (p,q,r,s) we've already processed
+        seen_2b: Dict[Tuple[int,int,int,int], None] = {}
+
+        for term, coeff in op.items():
+            # constant term: empty tuple
+            if not term:
+                constant = coeff.real
+                continue
+
+            # one‑body term: a†_σ,p a_σ,q  (σ = α or β)
+            if len(term) == 2:
+                (_, _, p), (_, _, q) = term
+                valid_1b = [(cre_a(p), des_a(q)), (cre_b(p), des_b(q))]
+                if term in valid_1b:
+                    one_body_tensor[p, q] += 0.5 * coeff.real
+                else:
+                    raise ValueError(f"Invalid one-body term {term}")
+                continue
+
+            # two‑body term: a†_σ,p a†_τ,r a_τ,s a_σ,q
+            if len(term) == 4:
+                (i1, i2, i3, i4) = term
+                _, _, p = i1
+                _, _, r = i2
+                _, _, s = i3
+                _, _, q = i4
+
+                valid_2b = [
+                    (cre_a(p), cre_a(r), des_a(s), des_a(q)),
+                    (cre_a(p), cre_b(r), des_b(s), des_a(q)),
+                    (cre_b(p), cre_a(r), des_a(s), des_b(q)),
+                    (cre_b(p), cre_b(r), des_b(s), des_b(q)),
+                ]
+                if term not in valid_2b:
+                    raise ValueError(f"Invalid two-body term {term}")
+
+                key = (p, q, r, s)
+                if key not in seen_2b:
+                    seen_2b[key] = None
+
+                    h_pqrs = 2.0 * coeff.real
```

Comment:
It looks like you are assuming the coefficients are real? They can be complex though.